les/peer.go

```diff
@@ -545,9 +545,11 @@ func (ps *peerSet) notify(n peerSetNotify) {
 func (ps *peerSet) Register(p *peer) error {
 	ps.lock.Lock()
 	if ps.closed {
+		ps.lock.Unlock()
```

Comment:
how about using "defer ps.lock.Unlock()"?

Reply:
I have thought about it.  but ps.lock.Unlock() should be called  before "for _, n := range peers {" in  561  line .
or  any other good way ?